### ISSUE TITLE
[BUG] SSP waveform error when fclk width > sample width

### ISSUE THREAD

**Issue by zrombel** [thesofproject/sof#2356]
**Describe the bug**
Bug regards configuration of FCLK pulse with of SSP interface. When setting frame_pulse_width > sample bit depth (for example sample bit depth = 16 and frame_pulse_width = 17, or sample bit depth = 32 and frame_pulse_width = 33) SSP wave forms are generated incorrectly: FCLK frequency changes and data bits are shifted. Bug occurs **only when channel count is greater then 1** and is sample rate independent. 

Example waveforms:

Audio Format: 8kHz 16/16b 2ch 
Frame pulse width: 1
Bclk: 512kHz
Fclk: 8kHz
Waveform: Correct
![image](https://user-images.githubusercontent.com/40820404/73937386-3741ef00-48e5-11ea-8799-a8faac42cc3b.png)

Audio Format: 8kHz 16/16b 2ch 
Frame pulse width: 16
Bclk: 512kHz
Fclk: 8kHz
Waveform: Correct
![image](https://user-images.githubusercontent.com/40820404/73937529-86881f80-48e5-11ea-9622-3a237a672c2b.png)

Audio Format: 8kHz 16/16b 2ch 
Frame pulse width: 17
Bclk: 512kHz
**Fclk: 6.4kHz**
Waveform: **Incorrect**
![image](https://user-images.githubusercontent.com/40820404/73937625-bdf6cc00-48e5-11ea-9ac7-96b41f3fe09d.png)

Audio Format: 8kHz 32/32b 2ch 
Frame pulse width: 1
Bclk: 1.2MHz
Fclk: 8kHz
Waveform: Correct
![image](https://user-images.githubusercontent.com/40820404/73937997-a409b900-48e6-11ea-850b-8e11e795864a.png)

Audio Format: 8kHz 32/32b 2ch 
Frame pulse width: 32
Bclk: 1.2MHz
Fclk: 8kHz
Waveform: Correct
![image](https://user-images.githubusercontent.com/40820404/73938096-d9aea200-48e6-11ea-9258-ce52c3c189d5.png)

Audio Format: 8kHz 32/32b 2ch 
Frame pulse width: 33
Bclk: 1.2MHz
**Fclk: 6.593kHz**
Waveform: **Incorrect**
![image](https://user-images.githubusercontent.com/40820404/73938274-33af6780-48e7-11ea-994a-04b9c06cd009.png)

**To Reproduce**
Python tests 01_06_TestPcm1bWaveformVerification with frame_pulse_width set accordingly to tested audio format.

**Reproduction Rate**
100%

**Environment**
Generic bug

**Logs**
[8kHz16b2ch1fpw.zip](https://github.com/thesofproject/sof/files/4165172/8kHz16b2ch1fpw.zip)
[8kHz16b2ch16fpw.zip](https://github.com/thesofproject/sof/files/4165173/8kHz16b2ch16fpw.zip)
[8kHz16b2ch17fpw.zip](https://github.com/thesofproject/sof/files/4165174/8kHz16b2ch17fpw.zip)
[8kHz32b2ch1fpw.zip](https://github.com/thesofproject/sof/files/4165175/8kHz32b2ch1fpw.zip)
[8kHz32b2ch32fpw.zip](https://github.com/thesofproject/sof/files/4165176/8kHz32b2ch32fpw.zip)
[8kHz32b2ch33fpw.zip](https://github.com/thesofproject/sof/files/4165177/8kHz32b2ch33fpw.zip)

**Comment by jajanusz** [comment]
@zrombel I guess it's on current master, right?

**Comment by zrombel** [comment]
Right. Found on master branch commit 0469a018ba856b0fd96b042c4011301e988d0f8f but it seems it's present in FW since beginning.

**Comment by plbossart** [comment]
IIRC there was a hardware "feature" that the pulse has to be limited to the first slot in master mode.
@lbetlej does this ring a bell? We had such requirements a long time ago from our automotive friends, I am not sure the SSP supports e.g. a TDM mode with a long pulse (or even half cycle)

**Comment by lgirdwood** [comment]
Pushed to v1.6 until further comments can be aligned.

**Comment by lgirdwood** [comment]
Ok, I guess we can live with this. Re-assign priority and milestone unless anyone objects.
